clients/agentic-sandbox-client-python/agentic_sandbox/sandbox.py

```diff
@@ -0,0 +1,188 @@
+                    is_ready = True
+                    break
+
+            if is_ready:
+                self.sandbox_name = sandbox_object['metadata']['name']
+                w.stop()
+                print(f"Sandbox {self.sandbox_name} is ready.")
+                break
+
+        if not self.sandbox_name:
+            self.__exit__(None, None, None)
+            raise TimeoutError("Sandbox did not become ready within the 180-second timeout period.")
+
+        print(f"Starting port-forwarding for sandbox {self.sandbox_name}...")
+        self.port_forward_process = subprocess.Popen(
+            [
+                "kubectl", "port-forward",
+                f"pod/{self.sandbox_name}",
+                f"{self.server_port}:{self.server_port}",
+                "-n", self.namespace
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        time.sleep(3)  # Give port-forward a moment to establish
+
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Deletes the SandboxClaim resource and stops port-forwarding."""
+        if self.port_forward_process:
+            print("Stopping port-forwarding...")
+            self.port_forward_process.terminate()
+            self.port_forward_process.wait()
+
+        if self.claim_name:
+            print(f"Deleting SandboxClaim: {self.claim_name}")
+            try:
+                self.custom_objects_api.delete_namespaced_custom_object(
+                    group=CLAIM_API_GROUP,
+                    version=CLAIM_API_VERSION,
+                    namespace=self.namespace,
+                    plural=CLAIM_PLURAL_NAME,
+                    name=self.claim_name
+                )
+            except client.ApiException as e:
+                if e.status != 404:
+                    print(f"Error deleting sandbox claim: {e}")
+
+    def run(self, command: str, timeout: int = 60) -> ExecutionResult:
+        """Executes a shell command inside the running sandbox."""
+        if not self.is_ready():
+            raise ConnectionError("Sandbox is not ready. Cannot execute commands.")
+
+        url = f"http://127.0.0.1:{self.server_port}/execute"
+        payload = {"command": command}
+
+        response = requests.post(url, json=payload, timeout=timeout)
+        response.raise_for_status()
+
+        response_data = response.json()
+        return ExecutionResult(
+            stdout=response_data['stdout'],
+            stderr=response_data['stderr'],
+            exit_code=response_data['exit_code']
+        )
+
+    def write(self, path: str, content: bytes | str):
+        """Uploads content to a file inside the sandbox."""
+        if not self.is_ready():
+            raise ConnectionError("Sandbox is not ready. Cannot write files.")
+
+        url = f"http://127.0.0.1:{self.server_port}/upload"
+        filename = os.path.basename(path)
+        files_payload = {'file': (filename, content)}
+
```

Comment:
The `read` method takes a `path` parameter, but it's not clear from the code or the docstring what the base path for the download is inside the sandbox. It would be helpful to clarify this.